zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -1074,11 +1117,58 @@ where
                 }
             }
             PoolType::Transparent => {
+                #[cfg(not(feature = "transparent-inputs"))]
                 return Err(Error::UnsupportedChangeType(output_pool));
             }
         }
     }
 
+    // This reserves the ephemeral addresses even if transaction construction fails.
+    // It is not worth the complexity of being able to unreserve them, because there
+    // are few failure modes after this point that would allow us to do so.
+    #[cfg(feature = "transparent-inputs")]
+    {
+        let ephemeral_outputs: Vec<(usize, &ChangeValue)> = proposal_step
+            .balance()
+            .proposed_change()
+            .iter()
+            .enumerate()
+            .filter(|(_, change_value)| {
+                change_value.is_ephemeral() && change_value.output_pool() == PoolType::Transparent
+            })
+            .collect();
+        if ephemeral_outputs.len() > i32::MAX as usize {
+            return Err(Error::ProposalNotSupported);
```

Comment:
Good to check this, but I think we have other problems at this point...

Reply:
True, perhaps we want to impose smaller limits? It prevents an arithmetic overflow though.